app/services/teacher_training_public_api/sync_sites.rb

```diff
@@ -17,17 +17,11 @@ def perform(provider_id, recruitment_cycle_year, course_id)
       ).includes(:location_status).paginate(per_page: 500)
 
       sites.each do |site_from_api|
-        site = provider.sites.find_or_create_by(code: site_from_api.code)
-
-        site.name = site_from_api.name
-        site.address_line1 = site_from_api.street_address_1&.strip
-        site.address_line2 = site_from_api.street_address_2&.strip
-        site.address_line3 = site_from_api.city&.strip
-        site.address_line4 = site_from_api.county&.strip
-        site.postcode = site_from_api.postcode&.strip
-        site.region = site_from_api.region_code&.strip
-        site.latitude = site_from_api.latitude
-        site.longitude = site_from_api.longitude
+        site = provider.sites.create_or_find_by(code: site_from_api.code) do |s|
+          s.name = site_from_api.name
```

Comment:
Maybe add a comment explaining that we need to set this so that the record is valid?